src/components/organisms/SettingsRead.tsx

```diff
@@ -0,0 +1,253 @@
+  },
+
+  defaultVariants: {
+    variant: "indigo",
+  },
+});
+const Fieldset = styled("fieldset", {
+  all: "unset",
+  marginBottom: 15,
+  width: "100%",
+  display: "flex",
+  flexDirection: "column",
+  justifyContent: "flex-start",
+});
+
+const Label = styled("label", {
+  fontSize: 13,
+  lineHeight: 1,
+  marginBottom: 10,
+  color: indigo.indigo12,
+  display: "block",
+});
+
+const Input = styled("input", {
+  all: "unset",
+  flex: "1 0 auto",
+  borderRadius: 4,
+  padding: "0 10px",
+  fontSize: 15,
+  lineHeight: 1,
+  color: indigo.indigo11,
+  boxShadow: `0 0 0 1px ${indigo.indigo7}`,
+  height: 35,
+  "&:focus": { boxShadow: `0 0 0 2px ${indigo.indigo8}` },
+  "&::placeholder": { color: mauve.mauve7 },
+});
+
+const FieldCheck = styled("fieldset", {
+  all: "unset",
+  width: "100%",
+  display: "flex",
+  alignItems: "center",
+});
+
+const CheckLink = styled("a", {
+  color: indigo.indigo11,
+  textDecoration: "underline",
+  "&:visited": { color: indigo.indigo11 },
+});
+
+type Props = {
+  setSettingsView: React.Dispatch<React.SetStateAction<number>>;
+  settingsView: React.Dispatch<React.SetStateAction<number>>;
+  selectedDeviceName: React.Dispatch<React.SetStateAction<string>>;
+  selectedDeviceIndex: React.Dispatch<React.SetStateAction<number>>;
+  deviceData: DeviceData;
+  userData: UserData;
+  isLoading: boolean;
+};
+
+const SettingsViewState = { DropDown: 0, Read: 1, Change: 2, Updated: 3 } as const; 
+type SettingViewState = typeof SettingsViewState[keyof typeof SettingsViewState];
+
+export default function SettingsRead({
+  setSettingsView,
+  settingsView,
+  selectedDeviceName,
+  selectedDeviceIndex,
+  deviceData,
+  userData,
+  isLoading,
+}: Props) {
+  // useEffect(() => {
+  //   let index: number = Number(selectedDeviceIndex);
+  //   console.log(deviceData[index]);
+  // }, []);
+
+  // console.log(settingsView);
+
+  // const setIndex = () => {
+  //   let index: number = Number(selectedDeviceIndex);
+  //   // let device: {} = dataArray[index];
+  //   console.log(deviceData[index]);
+  // };
+
+  // setIndex();
+
+  const openSettingsChange = () => {
+    setSettingsView(SettingsViewState.Change);
+  };
+
+  if (isLoading) {
+    return <div className="loading">Loading...</div>;
+  } else {
+    return (
+      <>
+        <div className={styles.top}>
+          <div className={styles.top__inner}>
+            <div className={styles.user_settings}>
+              <h2>User Settings</h2>
+              <br></br>
+              <Text>First Name: {userData.first_name}</Text>
+              <Text>Last Name: {userData.last_name}</Text>
+              <Text>Email: {userData.email}</Text>
+              <Text>Password: *****</Text>
+              <Text>Phone Number: {userData.phone_number}</Text>
+            </div>
+            <div className={styles.device_settings}>
+              <h2>Device Settings</h2>
+              <br></br>
+              <Text>Device ID: {deviceData[0].device_id}</Text>
```

Comment:
You can use `selectedDeviceIndex` instead of `0`

Reply:
Done